libs/initScripts/helpers/version.js

```diff
@@ -6,26 +6,45 @@ const web3 = require('web3')
  * @param {String} serviceType service type trying to register
  * @param {String} serviceVersionStr version string to register
  * @param {String?} privateKey optional private key string
+ * @param {String?} dryRun Optional parameter to return the generated parameters without sending tx
  */
-async function setServiceVersion (audiusLibs, serviceType, serviceVersionStr, privateKey = null) {
+async function setServiceVersion (audiusLibs, serviceType, serviceVersionStr, privateKey = null, dryRun = true) {
   if (!audiusLibs) throw new Error('audiusLibs is not defined')
 
   console.log('----version init---')
+  let resp
 
   try {
-    const resp = await audiusLibs.ethContracts.ServiceTypeManagerClient.setServiceVersion(
+    resp = await audiusLibs.ethContracts.ServiceTypeManagerClient.setServiceVersion(
       serviceType,
       serviceVersionStr,
-      privateKey)
-    console.log(resp)
+      privateKey,
+      dryRun
+    )
   } catch (e) {
     if (!e.toString().includes('Already registered')) {
       console.log(e)
+      return
     } else {
       console.log('Already registered')
     }
   }
 
+  // this prints out the fields and values to be submitted onto the Governance dashboard
+  // to create new proposals
+  if (dryRun) {
+    console.log(`
+      Set latest ${serviceType} version on chain to v${serviceVersionStr}
+      ServiceTypeManagerProxy
+      setServiceVersion(bytes32,bytes32)
+      ${resp}
+
+      Description:
+      Git SHA - <SHA>
```

Comment:
do you want this to print out actual git SHA?

Reply:
it's actually pretty difficult to fetch the sha since it's the SHA of the github release. didn't want to do a bunch of git fu since this is just supposed to print the template for the governance proposal. didnt seem worth it given the fact that we run this once every two weeks and it takes 10 seconds to look it up 🤷

Reply:
SHA of the release as in SHA of the tag? Makes sense, but copy-pasting SHA stuff is subject to human error (I have learned :() so I'd like to find a way to automate it down the line. I can try to add this via something like [simple-git](https://www.npmjs.com/package/simple-git) moving forward / help automate other parts of this process. My gut right now based on my understanding of the system suggests it's a good look for this type of public-facing process for it to be as deterministic and automated as possible even if the upfront cost of creating those conditions is a bit higher. My intuitions/mental model here might well be off though!